### ISSUE TITLE
Warrior: Recklessness, Devastate/Sunder Armor and Sword and Board

### ISSUE THREAD

**Issue by HeadOwns** [dalaranwow/dalaran-wow#542]
hello there, recently, I've noticed 3 bugs here, the 1st one is the Recklessness, updating what I've written before, the Recklessness bug is the following, if I have 3 stacks and my Whirlwind hits 2 or more targets, it will consume all the stacks but, it shows that it consumed only 1 stack, so 2 stacks left and they are kinda uselsess, so please, fix it, I mean, if you have 3 stacks and you hitted 2 targets with Whirlwind (2 times each target so 4 hits) the buff must go away, not stay and giving me the impression that I have 2 more stacks.

3rd one is Sword and Board, it doesn't happen to me always, but already did sometimes, when I get SaB proc, sometimes I can't use Shield Slam, it just turn around and consume a gcd from me but it doesn't hit it untill SaB goes away, and the 2nd SaB bug is that sometimes, when I get this proc and use Shield Slam, the proc stays with me till the seconds expire.

I hope u guys can fix these soon, regards.

Headscutter.

**Comment by robinsch** [comment]
Merged with https://github.com/dalaranwow/dalaran-wow/issues/1255

**Comment by Aridam** [comment]
Confirming the Recklessness bug as present on the live Realm to this day -- could use a fix to that.

**Comment by TeridanDW** [comment]
Hello, I can confirm the Recklessness bug too.
So reopening this report as the first bug is valid.

Easy steps to reproduce the problem:
1. Attack 3+ creatures
2. Use Recklessness
3. Use Whirlwind
4. Notice all damage is critical strikes - correct
5. Notice you have 2 stacks on Recklessness left
6. Use Whirlwind again
7. Notice not all attacks are criticals anymore - incorrect - they should be as the player still has 2 stacts on Recklessness

![image](https://user-images.githubusercontent.com/6797611/32702308-cfe4b20e-c7ed-11e7-9113-998f2695182a.png)

**Comment by Aridam** [comment]
Confirmed. This is a rather tedious and problematic bug as this gets your Recklessness stuck while being unable to land Crits with it, so you're forced to remove it manually as your only solid workaround -- minus the fact you pretty much throw what's left of your Recklessness away as well, so it's still half-assed even as a workaround.

**Comment by Aridam** [comment]
Update: Something I forgot to mention is that Recklessness is legit-stuck this way if you land 2+ hits, retaining the +% DMG taken, being able to use Enraged Regeneration, etc.

**Comment by Aridam** [comment]
Recklessness is still bugged as previously-confirmed in the prior posts.

**Comment by Rushor** [comment]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
When you use Slam under the effect of Recklessness (having at least 1 charge), you can do normal (yellow, not crit, not glancing, not blocked/parried/dodged/missed) strikes.
It doesn't seems to increase Slam critical chance but it does consume charges if slam crit.

**Expected behaviour**: Tell us what should happen instead
Recklessness add 100% critical chance to the next 3 ability attacks, slam being a normal melee ability, should be affected by it.
In the case I'm mistaken and, for some obscure reason, it should not be affected by Recklessness, then it should not consume charges.

**Steps to reproduce the problem**:

1. Get rid of all +crit items (to get low crit chance).
2. Find a mob with the same or lower lvl (to avoid problems with the target defense).
3. Use Recklessness and spam Slam.

**Include proofs for this behaviour**
Saw it happening on a dungeon and went to test it on a normal, same lvl, mob.
With less than 20% of crit chance I could get non crits with slam (a screenshot at the bottom).

Info:
https://wowgaming.altervista.org/aowow/?spell=1719#modifies:50+1+13+3
Slam is  in the list of "Modifies".
http://wowwiki.wikia.com/wiki/Recklessness?oldid=2297223

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**
1719

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**
https://i.imgur.com/WJFZLHz.jpg
You can see in the combat log the "normal" slams, between the start and end of Recklessness.

**Comment by Aridam** [comment]
Just-retested the for the sake of the report on Slam + Recklessness -- if Slam is done alone (meaning no Sweeping Strikes), Recklessness works 100% and you get 3x Slam Crits from it. However, I've just-done it _with_ Sweeping Strikes, and despite SS not being able to Crit (and not being meant to Crit), that _also_ bugged Recklessness out, and as thus I couldn't get any more Slam Crits; the Sweeps still weren't Crits by the way so at least that's not an extra bug to add to the mix.

So basically do no AoE/Cleave and not even Sweeping Strikes, else you screw your Recklessness up big-time.

**Comment by AlexCSCP** [comment]
@Aridam about the slam thing
Look at the screenshot (https://i.imgur.com/WJFZLHz.jpg), slam is done alone with no AoE nor Autoattack, and still doesn't crit.  So your theory of Recklessness being bugged by other skills is irrelevant when it comes to slam.

**Comment by Aridam** [comment]
https://imgur.com/a/tVQTonr

Slam Crits 1, 2, 3, -- check the Buffs: Reck dissipates correctly per Slam, and every Slam is a Crit, done just-within the full duration of Recklessness for the three charges to be all for Slam, and consumed successfully for every single one, and thus, every single one is a Crit, unlike yours.

So, for me, it's working -- only not if I ever use any form of AoE or Cleave attacks.

On another note, I'm Arms, whereas I've noticed that you're Fury -- perhaps there's something in Fury that bugs it out for you? I literally have Slam with Recklessness working 100% as Arms _except_ for whenever there's any AoE/Cleave Damage involved with it (even Sweeping Strikes, oddly-enough).

**Comment by AlexCSCP** [comment]
@Aridam 
Yes, I'm fury, so I can't get Sweeping Strikes (not enough talent points).

In the topic of slam testing, I unlearned all my talent, unequipped all items (except a weapon since I need one for the skill), loged out and in,  and went to hit an outland mob of my lvl.
(NSFW warning, I unequipped my pants)
first try https://imgur.com/Kk5I4lV
You can see 3 normal slam before I get killed and Recklessness fades.
second try (different mob) https://imgur.com/rQV3jUw
You can see 3 normal slam before Recklessness fades and nothing between the 2nd and 3rd.
third try (same conditions as second but Screenshot taken during the fight) 
You can see normal slams while Recklessness is still up.

This time, only the first 2 slam were critical. 
In the old screenshot you can see I got only the first slam critical, but there is a normal white attack critical too.
In both cases (equipped with talents and unequipped without talents) I use Recklessness, get 2 critical and then normal damage until Recklessness fades.

It's a pain to test Recklessness without talent and equip since it has long CD and I get killed by the mobs.

**Comment by Aridam** [comment]
I kept retesting a  few times and, finally, just-now it actually happened to me as well -- two Slam Crits and the third one was normal yet Recklessness stayed for its last seconds before fading on its own; guess I was having a luck streak with all-Slam-Crit periods during Recklessness all the time previously.

Will keep testing and see if I can do something else to see what _might_ be causing this, or if it's just some RNG case.

**Comment by Aridam** [comment]
Still a major mess of an issue.

**Comment by Aridam** [comment]
Still fakkin' buggered m9.
